File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> ans;
        queue<TreeNode*>Q;
        if(root!=NULL){
        Q.push(root);
        while(!Q.empty()){
            int size=Q.size();
            vector<int> tmp;
            for(int i=0;i<size;i++){
                TreeNode* curr=Q.front();
                tmp.push_back(curr->val);
                if(curr->left!=NULL) Q.push(curr->left);
                if(curr->right!=NULL) Q.push(curr->right);
                Q.pop();
            }
            ans.push_back(tmp);
        }}
        return ans;
    }
};

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public IList<IList<int>> LevelOrder(TreeNode root) {
        List<IList<int>> ans=new List<IList<int>>();
        if(root!=null){
            Queue<TreeNode> q = new Queue<TreeNode>();
            q.Enqueue(root);
            while(q.Count!=0){
                List<int> tmp=new List<int>();
                int count=q.Count;
                for(int i=0;i<count;i++){
                    TreeNode x = q.Peek();
                    q.Dequeue();
                    tmp.Add(x.val);
                    if(x.left!=null){
                        q.Enqueue(x.left);
                    }
                    if(x.right!=null){
                        q.Enqueue(x.right);
                    }
                }
                ans.Add(tmp);
            }
        }
        return ans;
    }
}